packages/web-scripts/src/Tasks/PreinstallTasks.ts

```diff
@@ -0,0 +1,59 @@
+import { default as spawn } from 'cross-spawn-promise';
+import { default as Debug } from 'debug';
+
+import { PreinstallTaskDesc } from '../SharedTypes';
+import { CONSUMING_ROOT } from '../Paths';
+
+const dbg = Debug('web-scripts:preinstall'); // eslint-disable-line new-cap
+
+export async function preinstallTask(
+  task: PreinstallTaskDesc,
+): Promise<string[]> {
+  const fns = [yarnRun];
+
+  return await Promise.all(
+    fns.map(async fn => {
+      dbg('Beginning %s task', fn.name);
+      const stdout = await fn(task);
+      dbg('Finished %s task', fn.name);
+      return stdout;
+    }),
+  );
+}
+
+/**
+ * This task will run yarn audit at the location from which it was called.
+ * Yarn audit returns a status code which is the sum of the following masks:
+ * 1 for INFO
+ * 2 for LOW
+ * 4 for MODERATE
+ * 8 for HIGH
+ * 16 for CRITICAL
+ *
+ * The threshold ceiling is therefore 31. The default value of the threshold
+ * is 32, returning 0 exit status unless a threshold option is set.
+ * @see https://yarnpkg.com/lang/en/docs/cli/audit/
+ */
+async function yarnRun(task: PreinstallTaskDesc): Promise<string> {
+  const cmd = 'npx';
+  const { threshold = 32 } = task;
```

Comment:
my only quibble here - this does require that people do a bit of learning about the underlying command. It's true of many other web-scripts commands so it's fine... can we think of anything better? is `--threshold warning` any better? should both be supported?

Also, a note - isn't the type of `--threshold` going to be `string` every time? maybe we need to coerce it anyway to avoid mistakes later.

Reply:
> Also, a note - isn't the type of --threshold going to be string every time?

Correct, I had thought about this myself and decided to leave the implicit coercion to keep the logic terse and readable. Maybe commander has something to help here. Otherwise, I'll just cast it to a number.

Reply:
> is --threshold warning any better? should both be supported?

I think how yarn uses bit masking for the final evaluation of the audit command is kind of weird. I like the idea of giving more user-friendly. What do you think about restricting the input to the following enums?


```
* 1 for INFO
 * 2 for LOW
 * 4 for MODERATE
 * 8 for HIGH
 * 16 for CRITICAL
```

Reply:
I like that!